lib/datasource/terraform-module/index.ts

```diff
@@ -127,16 +81,63 @@ export async function getReleases({
     if (latestVersion) {
       latestVersion.releaseTimestamp = res.published_at;
     }
+
     logger.trace({ dep }, 'dep');
-    const cacheMinutes = 30;
-    await packageCache.set(cacheNamespace, pkgUrl, dep, cacheMinutes);
     return dep;
-  } catch (err) {
+  }
+
+  @cache({
+    namespace: 'terraform-service-discovery',
+    key: (registryUrl: string) => `${registryUrl}/.well-known/terraform.json`,
+    ttlMinutes: 1440,
+  })
+  async getTerraformServiceDiscoveryResult(
+    registryUrl: string
+  ): Promise<ServiceDiscoveryResult> {
+    const discoveryURL = `${registryUrl}/.well-known/terraform.json`;
+    const serviceDiscovery = (
+      await this.http.getJson<ServiceDiscoveryResult>(discoveryURL)
+    ).body;
+    return serviceDiscovery;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  override handleSpecificErrors(err: HttpError): void {
```

Comment:
@viceice What do you think of this [new language feature?](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html#override-and-the---noimplicitoverride-flag)

Reply:
i like it, it supresses a lot bugs from happening when refactoring

Reply:
This will  help for this eslint issues: https://github.com/typescript-eslint/typescript-eslint/issues/1103